application/src/main/java/org/opentripplanner/routing/algorithm/filterchain/filters/transit/RemoveTransitIfStreetOnlyIsBetter.java

```diff
@@ -15,10 +15,21 @@
  */
 public class RemoveTransitIfStreetOnlyIsBetter implements RemoveItineraryFlagger {
 
+  private static final Consumer<OptionalInt> IGNORE_SUBSCRIBER = i -> {};
+
   private final CostLinearFunction costLimitFunction;
+  private final OptionalInt streetOnlyCost;
```

Comment:
Same optional as field issue.

Reply:
I created a `RemoveTransitIfStreetOnlyIsBetterResults` record and removed optionals from fields